experimental/dds/tree2/src/test/shared-tree-core/editManager.spec.ts

```diff
@@ -636,32 +636,216 @@ describe("EditManager", () => {
 		];
 
 		describe("Local commit rebasing", () => {
-			for (const { rebasedEditCount, trunkEditCount } of scenarios) {
-				it(`Rebase ${rebasedEditCount} local commits over ${trunkEditCount} trunk commits`, () => {
+			for (const { rebasedEditCount: localCount, trunkEditCount: trunkCount } of scenarios) {
+				// This test leads to the following branching structure:
+				// (0)-(T1)-...-(Tc)
+				//  +----------------(L1)-...-(Lc)
+				// Note that the EditManager is first fed the local edits, then the trunk edits.
+				// We reset the rebase/invert/compose counters before T1 is received.
+				it(`Rebase ${localCount} local commits over ${trunkCount} trunk commits`, () => {
 					const rebaser = new NoOpChangeRebaser();
-					rebaseLocalEditsOverTrunkEdits(rebasedEditCount, trunkEditCount, rebaser);
-					assert.equal(rebaser.rebasedCount, trunkEditCount * rebasedEditCount ** 2);
-					assert.equal(rebaser.invertedCount, trunkEditCount * rebasedEditCount);
-					assert.equal(
-						rebaser.composedCount,
-						trunkEditCount * (rebasedEditCount * 2 + 1),
+					const run = rebaseLocalEditsOverTrunkEdits(
+						localCount,
+						trunkCount,
+						rebaser,
+						true,
 					);
+					rebaser.rebasedCount = 0;
+					rebaser.invertedCount = 0;
+					rebaser.composedCount = 0;
+					run();
+					const actual = {
+						rebased: rebaser.rebasedCount,
+						inverted: rebaser.invertedCount,
+						composed: rebaser.composedCount,
+					};
+					const expected = {
+						// For each trunk edit received (factor of T),
+						// we rebase each local edit (factor of L) over...
+						// - the inverse of each local edit before it: (L - 1) / 2
+						// - the new trunk edit: 1
+						// - the rebased version of each local edit before it: (L - 1) / 2
+						rebased: trunkCount * localCount * localCount,
+						// For each trunk edit received (factor of T),
+						// we invert...
+						// - each local edit once: L
+						inverted: trunkCount * localCount,
+						// For each trunk edit received (factor of T),
+						// we compose...
+						// - the inverse of each local edits: L
+						// - the trunk edit:  1
+						// - the rebased version of each local edits: L
+						// This compose is part of the code path updating the local branch.
+						composed: trunkCount * (localCount * 2 + 1),
+					};
+					assert.deepEqual(actual, expected);
 				});
 			}
 		});
-		describe("Peer commit rebasing", () => {
-			for (const { rebasedEditCount, trunkEditCount } of scenarios) {
-				it(`Rebase ${rebasedEditCount} peer commits over ${trunkEditCount} trunk commits`, () => {
+		describe("Peer commit rebasing for peer with fixed seq ref#", () => {
```

Comment:
I think the formatting would look nicer if there were line breaks between the tests.

Reply:
Addressing this in https://github.com/microsoft/FluidFramework/pull/18447